### PR TITLE
#46 Added Data Filter to Snowflake Writer

### PR DESCRIPTION
**Pull request by wtran63**
Implementation of issue #46 
Also includes fix for issue #42 

Added data filter fields to structs, terraform schema
Updated Create, Read, and Update functions

### CODE REVIEW

File: plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go [hunk 1]
```diff
@@ -52,6 +52,28 @@ func resourceKeboolaSnowflakeWriterTables() *schema.Resource {
...
+						},
+						"where_column": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "",
+						},
+						"where_operator": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "eq",
```
Comment (yongchiegui) [inline plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go:68]: Is "eq" an option for the operator? From editing the data filter through Keboola, I only see "in" and "not in".

File: plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go [hunk 2]
```diff
@@ -221,13 +251,24 @@ func resourceKeboolaSnowflakeWriterTablesRead(d *schema.ResourceData, meta inter
...
-			"primary_key": tableConfig.PrimaryKey,
+			"db_name":       	tableConfig.DatabaseName,
+			"export":        	tableConfig.Export,
+			"table_id":      	tableConfig.TableID,
+			"incremental": 	 	tableConfig.Incremental,
+			"primary_key": 		tableConfig.PrimaryKey,
+			"changed_since": 	storageInputTable.ChangedSince,
+			"where_column":		storageInputTable.WhereColumn,
+			"where_operator": 	storageInputTable.WhereOperator,
+			"where_values": 	storageInputTable.WhereValues,
```
Comment (yongchiegui) [inline plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go]: What happens when `storageInputTable`'s `ChangedSince`, `WhereColumn`, `WhereOperator`, and `WhereValues` fields were omitted/zero-valued during the Create? Any weird results when you apply an update?

File: plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go [hunk 3]
```diff
@@ -121,6 +143,14 @@ func resourceKeboolaSnowflakeWriterTablesCreate(d *schema.ResourceData, meta int
...
 		}
 
+		if val, ok:= config["changed_since"]; ok { storageTable.ChangedSince = val.(string)}
+		if val, ok:= config["where_column"]; ok { storageTable.WhereColumn = val.(string)}
+		if val, ok:= config["where_operator"]; ok { storageTable.WhereOperator = val.(string)}
+
+		if q := config["where_values"]; q != nil {
+			storageTable.WhereValues = AsStringArray(q.([]interface{}))
+		}
+
```
Comment (yongchiegui) [inline plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go:153]: What happens if `ChangedSince`, `WhereColumn`, `WhereOperator` and `WhereValues` are omitted/zero-valued during the Create/Update? Any weird results when you apply it?

File: plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go (hunk 1 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go:68]: Yeah, they're represented as the following:
- "in" = "eq"
- "not in" = "ne"

File: plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go (hunk 3 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_writer_tables.go:153]: If you look at the struct definition:
```
type SnowflakeWriterStorageTable struct {
	Source          string   `json:"source"`
	Destination     string   `json:"destination"`
	Columns         []string `json:"columns"`
	ChangedSince    string   `json:"changed_since,omitempty"`
	WhereColumn     string   `json:"where_column,omitempty"`
	WhereOperator   string   `json:"where_operator,omitempty"`
	WhereValues     []string `json:"where_values,omitempty"`
}
```

For the 4 variables in the struct you mentioned, if they are nil or undefined when encoding/decoding the JSON they will be omitted or left out.

**Pull request merged**